aten/src/ATen/native/Convolution.cpp

```diff
@@ -119,6 +123,28 @@ auto ConvParams::view1d_as_2d() -> void {
   }
 }
 
+bool ConvParams::use_cpu_depthwise3x3_winograd(
+    const at::Tensor& input, const at::Tensor& weight) const
+{
+#ifdef __ARM_NEON__
+  // Currently only 3x3 depthwise convolutions on tensors of float are supported.
+  return (input.ndimension() == 4) &&
+         (input.size(1) == groups) &&
+         (weight.ndimension() == 4 ) &&
+         (weight.size(0) % input.size(1) == 0) &&
+         (weight.size(2) == 3) &&
+         (weight.size(3) == 3) &&
+         (input.device().type() == c10::DeviceType::CPU) &&
+         (input.scalar_type() == at::kFloat) &&
+         (weight.device().type() == c10::DeviceType::CPU) &&
+         (weight.scalar_type() == at::kFloat) &&
```

Comment:
I think you need (stride[0] == 1) && (stride[1] == 1) - I can find these in the original Depthwise3x3ConvOp:
    OPERATOR_NEEDS_FEATURE(this->stride_h() == 1);
    OPERATOR_NEEDS_FEATURE(this->stride_w() == 1);

Without this check the numeric result seems to be wrong for MobileNetV2. After patching it the result seems correct.

Reply:
Thank you for pointing that out Jiakai.  I know it must have been painful to debug!